test/e2e/libpod_suite_test.go

```diff
@@ -0,0 +1,316 @@
+		podmanBinary = os.Getenv("PODMAN_BINARY")
+	}
+	conmonBinary := filepath.Join(cwd, "../../bin/conmon")
+	if os.Getenv("CONMON_BINARY") != "" {
+		conmonBinary = os.Getenv("CONMON_BINARY")
+	}
+	storageOptions := STORAGE_OPTIONS
+	if os.Getenv("STORAGE_OPTIONS") != "" {
+		storageOptions = os.Getenv("STORAGE_OPTIONS")
+	}
+
+	runCBinary := "/usr/bin/runc"
+	CNIConfigDir := "/etc/cni/net.d"
+
+	return PodmanTest{
+		PodmanBinary:        podmanBinary,
+		ConmonBinary:        conmonBinary,
+		CrioRoot:            filepath.Join(tempDir, "crio"),
+		CNIConfigDir:        CNIConfigDir,
+		RunCBinary:          runCBinary,
+		RunRoot:             filepath.Join(tempDir, "crio-run"),
+		StorageOptions:      storageOptions,
+		SignaturePolicyPath: filepath.Join(INTEGRATION_ROOT, "test/policy.json"),
+		ArtifactPath:        ARTIFACT_DIR,
+		TempDir:             tempDir,
+	}
+}
+
+//MakeOptions assembles all the podman main options
+func (p *PodmanTest) MakeOptions() []string {
+	return strings.Split(fmt.Sprintf("--root %s --runroot %s --runtime %s --conmon %s --cni-config-dir %s",
+		p.CrioRoot, p.RunRoot, p.RunCBinary, p.ConmonBinary, p.CNIConfigDir), " ")
+}
+
+// Podman is the exec call to podman on the filesystem
+func (p *PodmanTest) Podman(args []string) *PodmanSession {
+	podmanOptions := p.MakeOptions()
+	podmanOptions = append(podmanOptions, strings.Split(p.StorageOptions, " ")...)
+	podmanOptions = append(podmanOptions, args...)
+	fmt.Printf("Running: %s %s\n", p.PodmanBinary, strings.Join(podmanOptions, " "))
+	command := exec.Command(p.PodmanBinary, podmanOptions...)
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		Fail(fmt.Sprintf("unable to run podman command: %s", strings.Join(podmanOptions, " ")))
+	}
+	return &PodmanSession{session}
+}
+
+// Cleanup cleans up the temporary store
+func (p *PodmanTest) Cleanup() {
+	// Stop all containers
+	session := p.Podman([]string{"stop", "-a", "--timeout", "1"})
+	session.Wait()
+	// Remove all containers
+	session = p.Podman([]string{"rm", "-fa"})
+	session.Wait()
+	// Nuke tempdir
+	if err := os.RemoveAll(p.TempDir); err != nil {
+		fmt.Printf("%q\n", err)
+	}
+}
+
+// GrepString takes session output and behaves like grep. it returns a bool
+// if successful and an array of strings on positive matches
+func (s *PodmanSession) GrepString(term string) (bool, []string) {
+	var (
+		greps   []string
+		matches bool
+	)
+
+	for _, line := range strings.Split(s.OutputToString(), "\n") {
+		if strings.Contains(line, term) {
+			matches = true
+			greps = append(greps, line)
+		}
+	}
+	return matches, greps
+}
+
+// Pull Images pulls multiple images
+func (p *PodmanTest) PullImages(images []string) error {
+	for _, i := range images {
+		p.PullImage(i)
+	}
+	return nil
+}
+
+// Pull Image a single image
+// TODO should the timeout be configurable?
+func (p *PodmanTest) PullImage(image string) error {
+	session := p.Podman([]string{"pull", image})
+	session.Wait(60)
```

Comment:
Do we really need to wait/test the pull?  If the image doesn't get pulled, won't other parts of the test fail?  If we really need to wait, could we pull an image and then loop through podman ps outputs looking for the image?

Reply:
Nvm, I've since discovered the value is the time to wait before timing out, I thought it was wait for 60 seconds then check status.